code/controllers/subsystem/radiation.dm

```diff
@@ -10,83 +10,95 @@ SUBSYSTEM_DEF(radiation)
 
 /datum/controller/subsystem/radiation/fire(resumed)
 	while (processing.len)
-		var/datum/radiation_pulse_information/pulse_information = popleft(processing)
+		var/datum/radiation_pulse_information/pulse_information = processing[1]
 
 		var/datum/weakref/source_ref = pulse_information.source_ref
 		var/atom/source = source_ref.resolve()
 		if (isnull(source))
+			processing.Cut(1, 2)
 			continue
 
 		pulse(source, pulse_information)
 
 		if (MC_TICK_CHECK)
 			return
 
+		processing.Cut(1, 2)
+
 /datum/controller/subsystem/radiation/stat_entry(msg)
 	msg = "[msg] | Pulses: [processing.len]"
 	return ..()
 
 /datum/controller/subsystem/radiation/proc/pulse(atom/source, datum/radiation_pulse_information/pulse_information)
 	var/list/cached_rad_insulations = list()
+	var/list/cached_turfs_to_process = pulse_information.turfs_to_process.Copy()
```

Comment:
why copy this?

Reply:
To cache it into a list.

Reply:
It's dumber then that. Just setting a proc var to a datum var caches it and makes access times faster